examples/tutorials/analysis-3d/non_detected_source.py

```diff
@@ -0,0 +1,254 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+"""
+Computing flux upper limits
+============
+
+Explore how to deal with flux upper limits for a non-detected source.
+
+Prerequisites
+-------------
+
+It is advisable to understand the general Gammapy modelling and fitting
+framework before proceeding with this notebook, eg doc:`docs/user-guide/modeling`.
+
+Context
+-------
+
+In the study of VHE sources, one often encounters no significant
+detection even after long exposures. In that case, it may be useful to
+compute flux upper limits for the said target consistent with the observation.
+
+Proposed approach
+-----------------
+
+In this section, we will use an empty observation from the H.E.S.S. DL3 DR1 to understand how
+to quantify non-detections. We will
+- Compute excess and significance maps
+- Perform a source model fit and do a likelihood ratio test
+- Compute differential upper limits
+- Compute integral upper limits
+- Look at the differential sensitivity of our observation/analysis
```

Comment:
Wouldn't it be clearer to split to scientific problematic into different cases:
- Test the presence of emission from a potential source with known position and morphology and compute integral and differential UL
- Test for the presence of emission anywhere in a map and compute an integral flux UL at any position (i.e. UL map). But beware that ULs given by a correlated excess map are NOT a source UL as one would need to apply a containment fraction correction.